test/unit/CMakeLists.txt

```diff
@@ -9,65 +9,52 @@ endif()
 
 if(USE_EXTERNAL_GTEST)
 
-if(NOT GTest_FOUND)
+  if(NOT GTest_FOUND)
     message(FATAL_ERROR "External GTest >= 1.8.1 not found")
-endif()
-message(STATUS "Using external GTest")
+  endif()
+  message(STATUS "Using external GTest")
 
-# CMake < 3.20.0 uses GTest::GTest
-# CMake >= 3.20 uses GTest::gtest, and deprecates GTest::GTest
-# so for older CMake, create an alias from GTest::GTest to GTest::gtest
-if(NOT TARGET GTest::gtest)
+  # CMake < 3.20.0 uses GTest::GTest
+  # CMake >= 3.20 uses GTest::gtest, and deprecates GTest::GTest
+  # so for older CMake, create an alias from GTest::GTest to GTest::gtest
+  if(NOT TARGET GTest::gtest)
     add_library(GTest::gtest INTERFACE IMPORTED)
     set_target_properties(GTest::gtest PROPERTIES
                           INTERFACE_LINK_LIBRARIES "GTest::GTest")
-endif()
+  endif()
 
 else()
 
-message(STATUS "Using internal GTest")
+  message(STATUS "Fetching GTest")
 
-#
-# Build Google Test
-#
-# Source https://github.com/google/googletest/blob/master/googletest/README.md
-# Download and unpack googletest at configure time
-configure_file(
-  ${PROJ_SOURCE_DIR}/test/googletest/CMakeLists.txt.in
-  ${PROJ_BINARY_DIR}/googletest-download/CMakeLists.txt)
-
-if(CYGWIN)
-# needed at least with gcc 11.3.0 Cygwin, otherwise build errors in
-# in googletest-src/googletest/include/gtest/internal/gtest-port.h related to
-# fileno() and other functions, when building gtest or including it
-add_definitions(-D_GNU_SOURCE)
-endif()
+  # Add Google Test
```

Comment:
Do you know why we fetch directly in CMake instead of relying on something like a git submodule? I think git submodule could be simpler/cleaner